tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,295 @@
+        private static readonly int __sampleSsnValue = 145014000;
+
+        private static BsonDocument SampleDoc =
+            new BsonDocument
+            {
+                { "name", __sampleNameValue },
+                { "ssn", __sampleSsnValue },
+                { "bloodType", "AB-" },
+                {
+                    "medicalRecords",
+                    new BsonArray(new []
+                    {
+                        new BsonDocument("weight", 180),
+                        new BsonDocument("bloodPressure", "120/80")
+                    })
+                },
+                {
+                    "insurance",
+                    new BsonDocument
+                    {
+                        { "policyNumber", 123142 },
+                        { "provider", "MaestCare" }
+                    }
+                }
+            };
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-insertdatawithencryptedfields-cs
+        [Fact]
+        public void InsertDataWithEncryptedFields()
+        {
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var medicalRecordsNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var base64KeyId = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", __sampleSsnValue);
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(base64KeyId);
+
+            // Construct an encrypted client
+            var encryptedClient = CreateEncryptedClient(
+                medicalRecordsNamespace,
+                keyVaultNamespace,
+                schema);
+            var collection = encryptedClient
+                .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+
+            // Insert a document into the collection
+            collection.UpdateOne(ssnQuery, new BsonDocument("$set", SampleDoc), new UpdateOptions() { IsUpsert = true });
+            Console.WriteLine("Successfully upserted the sample document!");
+
+            // Query SSN field with encrypted client
+            var result = collection.Find(ssnQuery).Single();
+
+            Console.WriteLine("Encrypted client query by the SSN (deterministically-encrypted) field:\n" + result.ToJson());
+
+            // Query SSN field with normal client without encryption
+            var normalMongoClient = new MongoClient(__connectionString);
+            collection = normalMongoClient
+              .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+              .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+            var normalClientResult = collection.Find(ssnQuery).FirstOrDefault();
```

Comment:
Can we just reuse `result` instead of defining every longer result variable names?